src/commands/cmd_bulk_insert.c

```diff
@@ -102,28 +109,23 @@ void _MGraph_BulkInsert(void *args) {
     // Disable matrix synchronization for bulk insert operation
     Graph_SetMatrixPolicy(gc->g, RESIZE_TO_CAPACITY);
 
-    if (argc == 0) {
-        // Only the allocation string was received, so our work is done.
-        // (any non-BEGIN query with an argc of 0 here would have failed the arity check)
-        len = snprintf(reply, 1024, "Initialized a graph to accommodate %lld nodes and %lld edges.", final_node_count, final_edge_count);
-        RedisModule_ReplyWithStringBuffer(ctx, reply, len);
-        goto cleanup;
-    }
+    // Allocate or extend datablocks to accommodate all incoming entities
+    Graph_AllocateNodes(gc->g, nodes_in_query + initial_node_count);
 
-    int rc = BulkInsert(ctx, gc, &nodes, &edges, argv, argc);
+    int rc = BulkInsert(ctx, gc, argv, argc);
 
     if (rc == BULK_FAIL) {
         // If insertion failed, clean up keyspace and free added entities.
         key = RedisModule_OpenKey(ctx, rs_graph_name, REDISMODULE_WRITE);
         RedisModule_DeleteKey(key);
-        goto cleanup;
-    } else if (rc == BULK_COMPLETE) {
+        gc = NULL;
```

Comment:
Leak when gc is created ?

Reply:
`RedisModule_DeleteKey` frees all data associated with the GraphContext. We set the pointer to NULL here because it's referring to memory we no longer own, and don't want to try to access it and release the lock in cleanup.